integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/ObservationFhirResourceProviderIntegrationTest.java

```diff
@@ -1365,6 +1372,153 @@ public void shouldReturnLastnEncountersObservationsWhenNoParamterIsGivenAsXml()
+		
+		assertThat(updatedObservation, notNullValue());
+		assertThat(updatedObservation.getIdElement().getIdPart(), not(equalTo(OBS_UUID)));
+		assertThat(updatedObservation.getCode().getCodingFirstRep().getCode(), is("5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"));
+		assertThat(updatedObservation, validResource());
+	}
+	
+	@Test
+	public void shouldReturnNotFoundWhenUpdatingNonExistentObservationAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Observation/" + OBS_UUID).accept(FhirMediaTypes.JSON).go();
+		Observation observation = readResponse(response);
+		
+		observation.setId(WRONG_OBS_UUID);
+		
+		response = put("/Observation/" + WRONG_OBS_UUID).jsonContent(toJson(observation)).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldReturnBadRequestWhenDocumentIdDoesNotMatchObservationIdAsJson() throws Exception {
+		MockHttpServletResponse response = get("/Observation/" + OBS_UUID).accept(FhirMediaTypes.JSON).go();
+		Observation observation = readResponse(response);
+		
+		observation.setId(WRONG_OBS_UUID);
+		
+		response = put("/Observation/" + OBS_UUID).jsonContent(toJson(observation)).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isBadRequest());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		OperationOutcome operationOutcome = readOperationOutcome(response);
+		
+		assertThat(operationOutcome, notNullValue());
+		assertThat(operationOutcome.hasIssue(), is(true));
+	}
+	
+	@Test
+	public void shouldUpdateExistingObservationAsXml() throws Exception {
+		String xmlObs;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(OBS_XML_UPDATE_PATH)) {
+			Objects.requireNonNull(is);
+			xmlObs = inputStreamToString(is, UTF_8);
+		}
+		
+		MockHttpServletResponse response = put("/Observation/" + OBS_UUID).xmlContent(xmlObs).accept(FhirMediaTypes.XML)
+		        .go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Observation updatedObservation = readResponse(response);
+		
+		assertThat(updatedObservation, notNullValue());
+		assertThat(updatedObservation.getIdElement().getIdPart(), not(equalTo(OBS_UUID)));
+		assertThat(updatedObservation.getCode().getCodingFirstRep().getCode(), is("5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"));
+		assertThat(updatedObservation, validResource());
+	}
+	
+	@Test
+	public void shouldPatchExistingObservationUsingJsonMergePatch() throws Exception {
```

Comment:
1. This should go after all the update methods
2. Ideally we'd have one for each type of patch, since that should already be supported.

Reply:
done!